fem/bilininteg_diffusion_pa.cpp

```diff
@@ -80,28 +80,35 @@ static void OccaPADiffusionSetup3D(const int D1D,
 }
 #endif // MFEM_USE_OCCA
 
+/*
 // PA Diffusion Assemble 2D kernel
 template<const int T_SDIM>
 static void PADiffusionSetup2D(const int Q1D,
-                               const int NE,
-                               const Array<double> &w,
-                               const Vector &j,
-                               const Vector &c,
-                               Vector &d);
+                             const int coeffDim,
+                             const int NE,
+                             const Array<double> &w,
+                             const Vector &j,
+                             const Vector &c,
+                             Vector &d);
+*/
+
 template<>
 void PADiffusionSetup2D<2>(const int Q1D,
+                           const int coeffDim,
                            const int NE,
                            const Array<double> &w,
                            const Vector &j,
                            const Vector &c,
                            Vector &d)
 {
    const int NQ = Q1D*Q1D;
+   const bool symmetric = (coeffDim != 4);
    const bool const_c = c.Size() == 1;
    auto W = w.Read();
    auto J = Reshape(j.Read(), NQ, 2, 2, NE);
-   auto C = const_c ? Reshape(c.Read(), 1, 1) : Reshape(c.Read(), NQ, NE);
-   auto D = Reshape(d.Write(), NQ, 3, NE);
+   auto C = const_c ? Reshape(c.Read(), 1, 1, 1) : Reshape(c.Read(), coeffDim,
```

Comment:
I think it's `coeffDim, 1 , 1` here for the first `Reshape`.

Reply:
If `const_c`, `c` is of size 1, so it only has entry `[0]`. This version passes the unit tests, so I think the dimensions are right.

Reply:
```suggestion
   auto C = const_c ? Reshape(c.Read(), coeffDim, 1, 1) : Reshape(c.Read(), coeffDim,
```

Reply:
What about a `ConstantMatrixCoefficient`?

Reply:
@YohannDudouit There are a few classes derived from `MatrixCoefficient` for which `EvalSymmetric` is not implemented, so the base class version will be called, resulting in an error. `MatrixConstantCoefficient` is one of those few classes. Support for those types can be added in the future as needed.